soccer/planning/planner/SettlePlanner.cpp

```diff
@@ -0,0 +1,177 @@
+#include "SettlePlanner.hpp"
+#include "motion/TrapezoidalMotion.hpp"
+#include "planning/trajectory/RRTUtil.hpp"
+#include "planning/trajectory/Trajectory.hpp"
+#include "planning/trajectory/VelocityProfiling.hpp"
+
+namespace Planning {
+ConfigDouble* SettlePlanner::_bufferTimeBeforeContact;
+ConfigDouble* SettlePlanner::_targetPointGain;
+ConfigDouble* SettlePlanner::_searchStartDist;
+ConfigDouble* SettlePlanner::_searchEndDist;
+ConfigDouble* SettlePlanner::_searchIncDist;
+ConfigDouble* SettlePlanner::_targetChangeThreshold;
+ConfigDouble* SettlePlanner::_shortcutDist;
+
+REGISTER_CONFIGURABLE(SettlePlanner);
+
+void SettlePlanner::createConfiguration(Configuration* cfg) {
+    _bufferTimeBeforeContact =
+        new ConfigDouble(cfg, "Capture/Settle/bufferTimeBeforeContact");
+    _targetPointGain = new ConfigDouble(cfg, "Capture/Settle/targetPointGain");
+    _searchStartDist = new ConfigDouble(cfg, "Capture/Settle/searchStartDist");
+    _searchEndDist = new ConfigDouble(cfg, "Capture/Settle/searchEndDist");
+    _searchIncDist = new ConfigDouble(cfg, "Capture/Settle/searchIncDist");
+    _targetChangeThreshold =
+        new ConfigDouble(cfg, "Capture/Settle/targetChangeThreshold");
+    _shortcutDist = new ConfigDouble(cfg, "Capture/Settle/shortcutDist");
+}
+using namespace Geometry2d;
+// todo(motion planning) find a better way to handle moving targets
+RJ::Time SettlePlanner::bruteForceIntercept(const PlanRequest& request) {
+    Ball& ball = request.context->state.ball;
+    std::vector<double> dists;
+    double endDist = *_searchEndDist;
+    const Rect& fieldRect = Field_Dimensions::Current_Dimensions.FieldRect();
+    Segment ballLine{ball.pos, ball.pos + ball.vel.normalized(endDist)};
+    // stop the search early if we go out of bounds
+    auto [isBoundaryHit, boundaryHits] = fieldRect.intersects(ballLine);
+    if (isBoundaryHit && boundaryHits.size() > 0) {
+        Point boundPoint = boundaryHits[0];
+        endDist = std::min(endDist, ball.pos.distTo(boundPoint));
+    }
+    for (double x = *_searchStartDist; x < endDist; x += *_searchIncDist) {
+        dists.push_back(x);
+    }
+    // brute force to find the path with the soonest contact time
+    RJ::Time interceptTime =
+        ball.estimateTimeTo(ball.pos + ball.vel.normalized(endDist));
+    _pathTargetPlanner.drawRadius = Ball_Radius;
+    _pathTargetPlanner.drawColor = Qt::white;
+    _pathTargetPlanner.drawLayer = "Settle Brute Force";
+#pragma omp parallel for default(none) \
+    shared(request, dists, ball, interceptTime)
+    for (int i = 0; i < dists.size(); i++) {
+        RJ::Time contactTime =
+            ball.estimateTimeTo(ball.pos + ball.vel.normalized(dists[i]));
```

Comment:
I'd expect that a surprisingly heft amount of the time here is being taken by `estimateTimeTo`. You could probably improve this by searching over intercept _times_ instead of intercept _positions_, and then doing forward-lookups instead of reverse lookups like this.